opensrp-pnc/src/main/java/org/smartregister/pnc/processor/PncMiniClientProcessorForJava.java

```diff
@@ -48,12 +67,185 @@ public boolean canProcess(@NonNull String eventType) {
+        generateKeyValuesFromEvent(event, keyValues);
+        String strStillBorn = keyValues.get(PncConstants.JsonFormKeyConstants.BABIES_STILL_BORN_MAP);
+        processStillBorn(strStillBorn, event);
+        String strBabiesBorn = keyValues.get(PncConstants.JsonFormKeyConstants.BABIES_BORN_MAP);
+        processBabiesBorn(strBabiesBorn, event);
+
+        PncDetails maternityDetails = new PncDetails(eventClient.getClient().getBaseEntityId(), event.getEventDate().toDate(), keyValues);
+        maternityDetails.setCreatedAt(new Date());
+        PncLibrary.getInstance().getPncOutcomeDetailsRepository().saveOrUpdate(maternityDetails);
+    }
+
+    private void processBabiesBorn(@Nullable String strBabiesBorn, @NonNull Event event) {
+        if (StringUtils.isNotBlank(strBabiesBorn)) {
+            try {
+                JSONObject jsonObject = new JSONObject(strBabiesBorn);
+                Iterator<String> repeatingGroupKeys = jsonObject.keys();
+                while (repeatingGroupKeys.hasNext()) {
+                    JSONObject jsonTestObject = jsonObject.optJSONObject(repeatingGroupKeys.next());
+                    PncChild pncChild = new PncChild();
+                    pncChild.setMotherBaseEntityId(event.getBaseEntityId());
+                    pncChild.setApgar(jsonTestObject.optString("apgar"));
+                    pncChild.setBfFirstHour(jsonTestObject.optString("bf_first_hour"));
+                    pncChild.setComplications(jsonTestObject.optString("baby_complications"));
+                    pncChild.setComplicationsOther(jsonTestObject.optString("baby_complications_other"));
+                    pncChild.setCareMgt(jsonTestObject.optString("baby_care_mgt"));
+                    pncChild.setFirstCry(jsonTestObject.optString("baby_first_cry"));
+                    pncChild.setDischargedAlive(jsonTestObject.optString("discharged_alive"));
+                    pncChild.setDob(jsonTestObject.optString("baby_dob"));
+                    pncChild.setFirstName(jsonTestObject.optString("baby_first_name"));
+                    pncChild.setLastName(jsonTestObject.optString("baby_last_name"));
+                    pncChild.setGender(jsonTestObject.optString("baby_gender"));
+                    pncChild.setChildHivStatus(jsonTestObject.optString("child_hiv_status"));
+                    pncChild.setHeight(jsonTestObject.optString("birth_height_entered"));
+                    pncChild.setWeight(jsonTestObject.optString("birth_weight_entered"));
+                    pncChild.setEventDate(PncUtils.convertDate(event.getEventDate().toDate(), PncDbConstants.DATE_FORMAT));
+                    pncChild.setNvpAdministration(jsonTestObject.optString("nvp_administration"));
+                    pncChild.setInterventionSpecify(jsonTestObject.optString("baby_intervention_specify"));
+                    pncChild.setInterventionReferralLocation(jsonTestObject.optString("baby_intervention_referral_location"));
```

Comment:
add these strings to constants and use its references here

Reply:
I have updated the file according to the need in issue 11 branch